File: android/quenderin-core/src/main/kotlin/ai/quenderin/core/ConversationExporter.kt
package ai.quenderin.core

/**
 * Renders a chat transcript to a portable Markdown document the user can share or save — their data
 * stays theirs, exportable off the device on THEIR terms (not silently uploaded). Pure + testable.
 * Twin of Swift `ConversationExporter`.
 */
object ConversationExporter {
    fun markdown(messages: List<ChatMessage>, title: String? = null): String {
        val heading = title?.trim().takeUnless { it.isNullOrEmpty() } ?: "Conversation"
        val count = messages.size
        val sb = StringBuilder()
        sb.append("# ").append(heading).append("\n\n")
        sb.append("_Exported from Quenderin — on-device, ").append(count)
            .append(" message").append(if (count == 1) "" else "s").append("._\n\n")
        for (m in messages) {
            val speaker = if (m.role == Role.USER) "You" else "Quenderin"
            sb.append("**").append(speaker).append(":**\n").append(m.text).append("\n\n")
        }
        return sb.toString().trim() + "\n"
    }
}

File: apple/QuenderinKit/Sources/QuenderinKit/ConversationExporter.swift
import Foundation

/// Renders a chat transcript to a portable Markdown document the user can share or save — their data
/// stays theirs, exportable off the device on THEIR terms (not silently uploaded). Pure + testable.
/// Twin of Android `ConversationExporter`.
public enum ConversationExporter {
    public static func markdown(_ messages: [ChatMessage], title: String? = nil) -> String {
        let trimmed = title?.trimmingCharacters(in: .whitespacesAndNewlines)
        let heading = (trimmed?.isEmpty == false) ? trimmed! : "Conversation"
        let count = messages.count
        var out = "# \(heading)\n\n"
        out += "_Exported from Quenderin — on-device, \(count) message\(count == 1 ? "" : "s")._\n\n"
        for message in messages {
            let speaker = message.role == .user ? "You" : "Quenderin"
            out += "**\(speaker):**\n\(message.text)\n\n"
        }
        return out.trimmingCharacters(in: .whitespacesAndNewlines) + "\n"
    }
}
